### DISCUSSION TITLE
Add physics to animation from blender ?

### DISCUSSION THREAD

**Discussion by Franckapik** [pmndrs/use-cannon#70] [General] [answered]
Is it possible to add physics to an object which is animated from blender to allow collision between meshes ? 
I use gtlftojsx and i would like to know if i have to "separate" mesh with blender animation and mesh that i have to animate only with r3f useFrame ... 

Thank you for your great work!
F.

**Comment by stockhuman** [comment, marked as answer]
That would be an amazing feature - I think it might surpass the capabilities of just about every game engine if we could have morphing collision meshes! Unfortunately, I don't think that's on the horizon. 

I've wanted to make a _cannon'd_ snake model that internally uses the same sort of constraints as in the [chain demo](https://github.com/react-spring/use-cannon/blob/master/examples/src/demos/Chain.js), but I don't think that's what you're suggesting. If you are asking if it is possible to animate a glTF import and to add a physics mesh to it, then yes, that is possible.

Here's a quick illustration of glTF animation and physics, using the cannon ping pong and r3f text demos.

https://codesandbox.io/s/a-more-cruel-ping-pong-4geih?file=/src/Bird.js
